File: src/static/js/api-config.js
// ARQV30 Enhanced v3.0 - API Configuration System
console.log('⚙️ Sistema de Configuração de APIs carregado');

class ApiConfigurationManager {
    constructor() {
        this.apis = {
            'gemini': { name: 'Google Gemini', env_var: 'GEMINI_API_KEY', critical: true },
            'openai': { name: 'OpenAI GPT', env_var: 'OPENAI_API_KEY', critical: true },
            'groq': { name: 'Groq Llama', env_var: 'GROQ_API_KEY', critical: false },
            'exa': { name: 'Exa Search', env_var: 'EXA_API_KEY', critical: true },
            'jina': { name: 'Jina Reader', env_var: 'JINA_API_KEY', critical: true },
            'youtube': { name: 'YouTube API', env_var: 'YOUTUBE_API_KEY', critical: false },
            'twitter': { name: 'Twitter API', env_var: 'TWITTER_BEARER_TOKEN', critical: false },
            'google_search': { name: 'Google Search', env_var: 'GOOGLE_SEARCH_KEY', critical: false }
        };
        
        this.configuredApis = new Set();
        this.init();
    }
    
    init() {
        console.log('🔧 Inicializando gerenciador de configuração de APIs');
        this.loadCurrentStatus();
    }
    
    async loadCurrentStatus() {
        try {
            const response = await fetch('/api/get_api_config');
            const data = await response.json();
            
            if (data.success) {
                this.updateConfiguredApis(data.config);
                this.updateOverviewStats();
            }
        } catch (error) {
            console.error('Erro ao carregar status das APIs:', error);
        }
    }
    
    updateConfiguredApis(config) {
        this.configuredApis.clear();
        
        Object.keys(config).forEach(api => {
            if (config[api]) {
                this.configuredApis.add(api);
            }
        });
        
        // Atualiza status visual
        Object.keys(this.apis).forEach(api => {
            this.updateApiStatus(api, config[api] || false);
        });
    }
    
    updateApiStatus(apiName, isConfigured) {
        const statusElement = document.getElementById(`${apiName}-status`);
        if (statusElement) {
            if (isConfigured) {
                statusElement.textContent = 'Configurado';
                statusElement.className = 'api-status configured';
            } else {
                statusElement.textContent = 'Não Configurado';
                statusElement.className = 'api-status missing';
            }
        }
    }
    
    updateOverviewStats() {
        const totalApis = Object.keys(this.apis).length;
        const configuredCount = this.configuredApis.size;
        const healthScore = Math.round((configuredCount / totalApis) * 100);
        
        const configuredElement = document.getElementById('configuredCount');
        const missingElement = document.getElementById('missingCount');
        const healthElement = document.getElementById('healthScore');
        
        if (configuredElement) configuredElement.textContent = configuredCount;
        if (missingElement) missingElement.textContent = totalApis - configuredCount;
        if (healthElement) healthElement.textContent = `${healthScore}%`;
    }
    
    async testApi(apiName, apiKey) {
        try {
            const response = await fetch('/api/save_api_config', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify({
                    api_name: apiName,
                    api_key: apiKey
                })
            });
            
            const result = await response.json();
            
            if (result.success) {
                this.configuredApis.add(apiName);
                this.updateApiStatus(apiName, true);
                this.updateOverviewStats();
                return { success: true, message: result.message };
            } else {
                return { success: false, error: result.error };
            }
            
        } catch (error) {
            return { success: false, error: error.message };
        }
    }
    
    async testAllApis() {
        const results = {};
        
        for (const apiName of Object.keys(this.apis)) {
            const keyInput = document.getElementById(`${apiName}-key`);
            if (keyInput && keyInput.value.trim()) {
                results[apiName] = await this.testApi(apiName, keyInput.value.trim());
            }
        }
        
        return results;
    }
    
    getConfigurationStatus() {
        const totalApis = Object.keys(this.apis).length;
        const configuredCount = this.configuredApis.size;
        const criticalApis = Object.values(this.apis).filter(api => api.critical).length;
        const criticalConfigured = Object.keys(this.apis)
            .filter(key => this.apis[key].critical && this.configuredApis.has(key)).length;
        
        return {
            total: totalApis,
            configured: configuredCount,
            missing: totalApis - configuredCount,
            critical_total: criticalApis,
            critical_configured: criticalConfigured,
            health_score: Math.round((configuredCount / totalApis) * 100),
            critical_health: Math.round((criticalConfigured / criticalApis) * 100),
            ready_for_analysis: criticalConfigured >= Math.ceil(criticalApis * 0.7) // 70% das críticas
        };
    }
}

// Global functions for API configuration
async function testAndSaveApi(apiName) {
    const keyInput = document.getElementById(`${apiName}-key`);
    const apiKey = keyInput.value.trim();
    
    if (!apiKey) {
        showNotification('Por favor, insira a chave da API', 'warning');
        return;
    }
    
    const button = event.target;
    const originalContent = button.innerHTML;
    button.innerHTML = '<i class="fas fa-spinner fa-spin"></i> Testando...';
    button.disabled = true;
    
    try {
        const result = await apiConfigManager.testApi(apiName, apiKey);
        
        if (result.success) {
            showNotification(`✅ ${apiName.toUpperCase()} configurado com sucesso!`, 'success');
            keyInput.value = ''; // Limpa o campo por segurança
        } else {
            showNotification(`❌ Erro ao configurar ${apiName}: ${result.error}`, 'error');
        }
        
    } catch (error) {
        showNotification(`❌ Erro de conexão: ${error.message}`, 'error');
    } finally {
        button.innerHTML = originalContent;
        button.disabled = false;
    }
}

async function testAndSaveGoogleSearch() {
    const keyInput = document.getElementById('google-search-key');
    const cseInput = document.getElementById('google-cse-id');
    
    const apiKey = keyInput.value.trim();
    const cseId = cseInput.value.trim();
    
    if (!apiKey || !cseId) {
        showNotification('Por favor, insira tanto a chave quanto o CSE ID', 'warning');
        return;
    }
    
    const button = event.target;
    const originalContent = button.innerHTML;
    button.innerHTML = '<i class="fas fa-spinner fa-spin"></i> Testando...';
    button.disabled = true;
    
    try {
        // Testa Google Search Key
        const keyResult = await apiConfigManager.testApi('google_search', apiKey);
        
        if (keyResult.success) {
            // Salva CSE ID também
            await fetch('/api/save_api_config', {
                method: 'POST',
                headers: { 'Content-Type': 'application/json' },
                body: JSON.stringify({
                    api_name: 'google_cse',
                    api_key: cseId
                })
            });
            
            showNotification('✅ Google Custom Search configurado com sucesso!', 'success');
            keyInput.value = '';
            cseInput.value = '';
            
            // Atualiza status
            apiConfigManager.updateApiStatus('google-search', true);
            
        } else {
            showNotification(`❌ Erro ao configurar Google Search: ${keyResult.error}`, 'error');
        }
        
    } catch (error) {
        showNotification(`❌ Erro de conexão: ${error.message}`, 'error');
    } finally {
        button.innerHTML = originalContent;
        button.disabled = false;
    }
}

async function testAllApis() {
    const button = event.target;
    const originalContent = button.innerHTML;
    button.innerHTML = '<i class="fas fa-spinner fa-spin"></i> Testando Todas...';
    button.disabled = true;
    
    try {
        const results = await apiConfigManager.testAllApis();
        
        let successCount = 0;
        let totalTested = 0;
        
        Object.entries(results).forEach(([api, result]) => {
            totalTested++;
            if (result.success) {
                successCount++;
            }
        });
        
        if (successCount === totalTested) {
            showNotification(`✅ Todas as ${totalTested} APIs testadas com sucesso!`, 'success');
        } else {
            showNotification(`⚠️ ${successCount}/${totalTested} APIs configuradas com sucesso`, 'warning');
        }
        
        apiConfigManager.updateOverviewStats();
        
    } catch (error) {
        showNotification(`❌ Erro ao testar APIs: ${error.message}`, 'error');
    } finally {
        button.innerHTML = originalContent;
        button.disabled = false;
    }
}

function showApiGuide() {
    const guideContent = `
        <div class="api-guide">
            <h3>🔧 Guia de Configuração de APIs</h3>
            
            <div class="guide-section">
                <h4>1. APIs Críticas (Obrigatórias)</h4>
                <ul>
                    <li><strong>Google Gemini:</strong> Modelo principal de IA</li>
                    <li><strong>Exa Search:</strong> Busca neural avançada</li>
                    <li><strong>Jina Reader:</strong> Extração de conteúdo</li>
                </ul>
            </div>
            
            <div class="guide-section">
                <h4>2. APIs Importantes (Recomendadas)</h4>
                <ul>
                    <li><strong>OpenAI:</strong> Fallback de IA</li>
                    <li><strong>YouTube:</strong> Análise de vídeos</li>
                </ul>
            </div>
            
            <div class="guide-section">
                <h4>3. Ordem de Configuração Recomendada</h4>
                <ol>
                    <li>Configure Gemini primeiro (modelo primário)</li>
                    <li>Configure Exa Search (pesquisa principal)</li>
                    <li>Configure Jina Reader (extração de conteúdo)</li>
                    <li>Configure OpenAI (fallback)</li>
                    <li>Configure APIs sociais (opcional)</li>
                </ol>
            </div>
        </div>
    `;
    
    showModal('Guia de Configuração', guideContent);
}

function showApiPricing() {
    const pricingContent = `
        <div class="api-pricing">
            <h3>💰 Informações de Preços das APIs</h3>
            
            <div class="pricing-item">
                <h4>Google Gemini</h4>
                <p>Gratuito até 15 requisições/minuto</p>
                <p>Plano pago: $0.00025 por 1K caracteres</p>
            </div>
            
            <div class="pricing-item">
                <h4>OpenAI GPT-4</h4>
                <p>$0.03 por 1K tokens de entrada</p>
                <p>$0.06 por 1K tokens de saída</p>
            </div>
            
            <div class="pricing-item">
                <h4>Exa Search</h4>
                <p>1000 buscas gratuitas/mês</p>
                <p>Plano pago: $5/mês para 5K buscas</p>
            </div>
            
            <div class="pricing-item">
                <h4>Jina Reader</h4>
                <p>1M caracteres gratuitos/mês</p>
                <p>Plano pago: $20/mês para 20M caracteres</p>
            </div>
        </div>
    `;
    
    showModal('Preços das APIs', pricingContent);
}

function showTroubleshooting() {
    const troubleshootingContent = `
        <div class="troubleshooting">
            <h3>🔧 Solução de Problemas</h3>
            
            <div class="problem-solution">
                <h4>❌ "API Key inválida"</h4>
                <ul>
                    <li>Verifique se copiou a chave completa</li>
                    <li>Confirme se a API está ativada no painel</li>
                    <li>Verifique se há créditos disponíveis</li>
                </ul>
            </div>
            
            <div class="problem-solution">
                <h4>⚠️ "Quota excedida"</h4>
                <ul>
                    <li>Aguarde reset da quota (geralmente 24h)</li>
                    <li>Considere upgrade do plano</li>
                    <li>Use APIs alternativas temporariamente</li>
                </ul>
            </div>
            
            <div class="problem-solution">
                <h4>🔄 "Erro de conexão"</h4>
                <ul>
                    <li>Verifique sua conexão com internet</li>
                    <li>Tente novamente em alguns minutos</li>
                    <li>Verifique se não há firewall bloqueando</li>
                </ul>
            </div>
        </div>
    `;
    
    showModal('Solução de Problemas', troubleshootingContent);
}

function showModal(title, content) {
    // Remove modal existente se houver
    const existingModal = document.getElementById('helpModal');
    if (existingModal) {
        existingModal.remove();
    }
    
    const modal = document.createElement('div');
    modal.id = 'helpModal';
    modal.className = 'api-config-modal';
    modal.style.display = 'flex';
    
    modal.innerHTML = `
        <div class="api-config-content">
            <div class="api-config-header">
                <h3 class="api-config-title">${title}</h3>
                <button class="api-config-close" onclick="closeHelpModal()">
                    <i class="fas fa-times"></i>
                </button>
            </div>
            <div class="modal-body">
                ${content}
            </div>
        </div>
    `;
    
    document.body.appendChild(modal);
}

function closeHelpModal() {
    const modal = document.getElementById('helpModal');
    if (modal) {
        modal.remove();
    }
}

// Global instance
let apiConfigManager;

// Initialize when DOM is ready
document.addEventListener('DOMContentLoaded', function() {
    apiConfigManager = new ApiConfigurationManager();
    console.log('✅ Gerenciador de APIs inicializado');
});

// Export for use in other scripts
window.apiConfigManager = apiConfigManager;
window.testAndSaveApi = testAndSaveApi;
window.testAndSaveGoogleSearch = testAndSaveGoogleSearch;
window.testAllApis = testAllApis;
window.showApiGuide = showApiGuide;
window.showApiPricing = showApiPricing;
window.showTroubleshooting = showTroubleshooting;

File: src/routes/api_config.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
ARQV30 Enhanced v3.0 - API Configuration Routes
Rotas para configuração de APIs na interface
"""

import os
import logging
from datetime import datetime
from flask import Blueprint, request, jsonify, render_template
from services.api_configuration_manager import api_config_manager

logger = logging.getLogger(__name__)

api_config_bp = Blueprint('api_config', __name__)

@api_config_bp.route('/api_status')
def api_status_page():
    """Página de status das APIs"""
    try:
        # Testa todas as APIs
        test_results = api_config_manager.test_all_apis()
        
        # Prepara dados para o template
        apis_data = []
        
        # APIs de IA
        ai_apis = [
            {
                'name': 'Google Gemini',
                'provider': 'Google AI',
                'env_var': 'GEMINI_API_KEY',
                'description': 'Modelo primário de IA para análises ultra-detalhadas',
                'priority': 'critical',
                'usage': 'Análise principal, avatar, drivers mentais',
                'status': 'configured' if test_results['detailed_results']['gemini']['working'] else 'missing'
            },
            {
                'name': 'OpenAI GPT',
                'provider': 'OpenAI',
                'env_var': 'OPENAI_API_KEY', 
                'description': 'Modelo de fallback para análises complementares',
                'priority': 'high',
                'usage': 'Fallback de IA, análises específicas',
                'status': 'configured' if test_results['detailed_results']['openai']['working'] else 'missing'
            },
            {
                'name': 'Groq Llama',
                'provider': 'Groq',
                'env_var': 'GROQ_API_KEY',
                'description': 'Modelo de alta velocidade para processamento rápido',
                'priority': 'medium',
                'usage': 'Processamento rápido, fallback secundário',
                'status': 'configured' if test_results['detailed_results']['groq']['working'] else 'missing'
            }
        ]
        
        # APIs de Busca
        search_apis = [
            {
                'name': 'Exa Neural Search',
                'provider': 'Exa AI',
                'env_var': 'EXA_API_KEY',
                'description': 'Busca neural semântica avançada',
                'priority': 'high',
                'usage': 'Pesquisa web principal, busca semântica',
                'status': 'configured' if test_results['detailed_results']['exa']['working'] else 'missing'
            },
            {
                'name': 'Google Custom Search',
                'provider': 'Google',
                'env_var': 'GOOGLE_SEARCH_KEY',
                'description': 'Busca por palavras-chave no Google',
                'priority': 'medium',
                'usage': 'Busca complementar, dados específicos',
                'status': 'configured' if os.getenv('GOOGLE_SEARCH_KEY') else 'missing'
            },
            {
                'name': 'Jina Reader',
                'provider': 'Jina AI',
                'env_var': 'JINA_API_KEY',
                'description': 'Extração de conteúdo web avançada',
                'priority': 'high',
                'usage': 'Extração de conteúdo, leitura de páginas',
                'status': 'configured' if test_results['detailed_results']['jina']['working'] else 'missing'
            }
        ]
        
        # APIs Sociais
        social_apis = [
            {
                'name': 'YouTube Data API',
                'provider': 'Google',
                'env_var': 'YOUTUBE_API_KEY',
                'description': 'Análise de conteúdo do YouTube',
                'priority': 'medium',
                'usage': 'Análise de vídeos, tendências',
                'status': 'configured' if test_results['detailed_results']['youtube']['working'] else 'missing'
            },
            {
                'name': 'Twitter API',
                'provider': 'Twitter/X',
                'env_var': 'TWITTER_BEARER_TOKEN',
                'description': 'Análise de tweets e tendências',
                'priority': 'medium',
                'usage': 'Análise social, sentimentos',
                'status': 'configured' if test_results['detailed_results']['twitter']['working'] else 'missing'
            }
        ]
        
        # Combina todas as APIs
        all_apis = ai_apis + search_apis + social_apis
        
        # Calcula estatísticas
        total_apis = len(all_apis)
        configured_apis = len([api for api in all_apis if api['status'] == 'configured'])
        critical_missing = len([api for api in all_apis if api['priority'] == 'critical' and api['status'] != 'configured'])
        
        stats = {
            'configured': configured_apis,
            'missing': total_apis - configured_apis,
            'critical_missing': critical_missing,
            'health_percentage': (configured_apis / total_apis) * 100
        }
        
        return render_template('api_status.html', 
                             apis=all_apis, 
                             stats=stats,
                             timestamp=datetime.now().strftime('%d/%m/%Y %H:%M:%S'))
        
    except Exception as e:
        logger.error(f"❌ Erro na página de status: {e}")
        return render_template('api_status.html', 
                             apis=[], 
                             stats={'error': str(e)},
                             timestamp=datetime.now().strftime('%d/%m/%Y %H:%M:%S'))

@api_config_bp.route('/test_all_apis', methods=['POST'])
def test_all_apis():
    """Testa todas as APIs configuradas"""
    try:
        test_results = api_config_manager.test_all_apis()
        
        return jsonify({
            'success': True,
            'test_results': test_results,
            'timestamp': datetime.now().isoformat()
        })
        
    except Exception as e:
        logger.error(f"❌ Erro ao testar APIs: {e}")
        return jsonify({
            'success': False,
            'error': str(e)
        }), 500

@api_config_bp.route('/save_api_config', methods=['POST'])
def save_api_config():
    """Salva configuração de API"""
    try:
        data = request.get_json()
        
        if not data:
            return jsonify({'success': False, 'error': 'Dados não fornecidos'}), 400
        
        api_name = data.get('api_name')
        api_key = data.get('api_key')
        
        if not api_name or not api_key:
            return jsonify({'success': False, 'error': 'Nome da API e chave são obrigatórios'}), 400
        
        # Mapeia nome da API para variável de ambiente
        env_var_map = {
            'gemini': 'GEMINI_API_KEY',
            'openai': 'OPENAI_API_KEY',
            'groq': 'GROQ_API_KEY',
            'exa': 'EXA_API_KEY',
            'jina': 'JINA_API_KEY',
            'youtube': 'YOUTUBE_API_KEY',
            'twitter': 'TWITTER_BEARER_TOKEN',
            'google_search': 'GOOGLE_SEARCH_KEY',
            'google_cse': 'GOOGLE_CSE_ID'
        }
        
        env_var = env_var_map.get(api_name)
        if not env_var:
            return jsonify({'success': False, 'error': 'API não reconhecida'}), 400
        
        # Define variável de ambiente (temporariamente)
        os.environ[env_var] = api_key
        
        # Testa a API
        try:
            if api_name == 'gemini':
                import google.generativeai as genai
                genai.configure(api_key=api_key)
                model = genai.GenerativeModel("gemini-2.0-flash-exp")
                response = model.generate_content("Teste")
                test_success = bool(response.text)
            else:
                test_success = True  # Para outras APIs, assume sucesso
                
        except Exception as test_error:
            return jsonify({
                'success': False,
                'error': f'Teste da API falhou: {str(test_error)}'
            }), 400
        
        return jsonify({
            'success': True,
            'message': f'API {api_name} configurada e testada com sucesso',
            'test_success': test_success
        })
        
    except Exception as e:
        logger.error(f"❌ Erro ao salvar configuração: {e}")
        return jsonify({
            'success': False,
            'error': str(e)
        }), 500

@api_config_bp.route('/get_api_config', methods=['GET'])
def get_api_config():
    """Obtém configuração atual das APIs"""
    try:
        config = {
            'gemini': bool(os.getenv('GEMINI_API_KEY')),
            'openai': bool(os.getenv('OPENAI_API_KEY')),
            'groq': bool(os.getenv('GROQ_API_KEY')),
            'exa': bool(os.getenv('EXA_API_KEY')),
            'jina': bool(os.getenv('JINA_API_KEY')),
            'youtube': bool(os.getenv('YOUTUBE_API_KEY')),
            'twitter': bool(os.getenv('TWITTER_BEARER_TOKEN')),
            'google_search': bool(os.getenv('GOOGLE_SEARCH_KEY')),
            'google_cse': bool(os.getenv('GOOGLE_CSE_ID'))
        }
        
        return jsonify({
            'success': True,
            'config': config,
            'total_configured': sum(config.values()),
            'total_apis': len(config)
        })
        
    except Exception as e:
        logger.error(f"❌ Erro ao obter configuração: {e}")
        return jsonify({
            'success': False,
            'error': str(e)
        }), 500